### ISSUE TITLE
gnss fix does not get updated

### ISSUE THREAD

**Issue by patcmorneau** [CIDCO-dev/Poseidon#273]
when starting the raspberrypi it launch poseidon but gps fix is not there yet and does not get updated after

**Comment by glabmoris** [comment]
Can you explain the problem a bit more in detail? 

Do you get a fix and it doesn't get updated?

**Comment by patcmorneau** [comment]
thats right, once i see that the rpi date is sync with the gps i have to restart the poseidon stack... happened with the zf9p

**Comment by patcmorneau** [comment]
[rosmaster.master][INFO] 2022-02-16 14:58:08,977: -SERVICE [/state_controller/get_loggers] /state_controller rosrpc://ubuntu:52793
[rosmaster.threadpool][ERROR] 2022-02-16 14:58:08,979: Traceback (most recent call last):
  File "/opt/ros/noetic/lib/python3/dist-packages/rosmaster/threadpool.py", line 218, in run
    result = cmd(*args)
  File "/opt/ros/noetic/lib/python3/dist-packages/rosmaster/master_api.py", line 210, in publisher_update_task
    ret = xmlrpcapi(api).publisherUpdate('/master', topic, pub_uris)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1109, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1450, in __request
    response = self.__transport.request(
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1153, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1169, in single_request
    return self.parse_response(resp)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1341, in parse_response
    return u.close()
  File "/usr/lib/python3.8/xmlrpc/client.py", line 655, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -1: 'publisherUpdate: unknown method name'>

**Comment by patcmorneau** [comment]
i think the problem is in the gpsd_client package
with command cgps the status is 2
but in gpsd_client/src/client.cpp line 266 -> fix.status.status is empty (no status)
i believe theres something wrong with the switch case at line 231 in client.cpp

gpsd installed vesion:
gpsd is already the newest version (3.20-8ubuntu0.4).

```


      /* gpsmm pollutes the global namespace with STATUS_,
       * so we need to use the ROS message's integer values
       * for status.status
       */
      switch (p->status) {
        case STATUS_NO_FIX:
          fix->status.status = -1; // NavSatStatus::STATUS_NO_FIX;
          break;
        case STATUS_FIX:
          fix->status.status = 0; // NavSatStatus::STATUS_FIX;
          break;
// STATUS_DGPS_FIX was removed in API version 6 but re-added afterward
#if GPSD_API_MAJOR_VERSION != 6 
        case STATUS_DGPS_FIX:
          fix->status.status = 2; // NavSatStatus::STATUS_GBAS_FIX;
          break;
#endif
      }

      fix->status.service = NavSatStatus::SERVICE_GPS;

      fix->latitude = p->fix.latitude;
      fix->longitude = p->fix.longitude;
      fix->altitude = p->fix.altitude;

      /* gpsd reports status=OK even when there is no current fix,
       * as long as there has been a fix previously. Throw out these
       * fake results, which have NaN variance
       */
      if (std::isnan(p->fix.epx) && check_fix_by_variance) {
        return;
      }

      fix->position_covariance[0] = p->fix.epx;
      fix->position_covariance[4] = p->fix.epy;
      fix->position_covariance[8] = p->fix.epv;

      fix->position_covariance_type = NavSatFix::COVARIANCE_TYPE_DIAGONAL_KNOWN;

      navsat_fix_pub.publish(fix);
      ROS_INFO_STREAM("gpsd_client, fix status : " << fix->status.status);
    }
```

**Comment by patcmorneau** [comment]
gnss callback in logger text does not get called and cannot update bootstrapGnssTime, without gnss fix we cannot toggle the logging.

the reason why its not getting called : xmlrpc.client.Fault: <Fault -1: 'publisherUpdate: unknown method name'>
```
[rosmaster.master][INFO] 2022-02-17 16:54:20,147: publisherUpdate[/fix] -> http://ubuntu:36719/ []: sec=0.01, exception=<Fault -1: 'publisherUpdate: unknown method name'>
[rosmaster.threadpool][ERROR] 2022-02-17 16:54:20,148: Traceback (most recent call last):
  File "/opt/ros/noetic/lib/python3/dist-packages/rosmaster/threadpool.py", line 218, in run
    result = cmd(*args)
  File "/opt/ros/noetic/lib/python3/dist-packages/rosmaster/master_api.py", line 210, in publisher_update_task
    ret = xmlrpcapi(api).publisherUpdate('/master', topic, pub_uris)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1109, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1450, in __request
    response = self.__transport.request(
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1153, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1169, in single_request
    return self.parse_response(resp)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1341, in parse_response
    return u.close()
  File "/usr/lib/python3.8/xmlrpc/client.py", line 655, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -1: 'publisherUpdate: unknown method name'>
```
possible explanation :
https://github.com/ros/ros_comm/issues/572#issuecomment-89087064
